web/src/settings_emoji.ts

```diff
@@ -267,8 +283,11 @@ function show_modal(): void {
         const formData = new FormData();
         const files = $<HTMLInputElement>("input#emoji_file_input")[0]!.files;
         assert(files !== null);
-        for (const [i, file] of [...files].entries()) {
-            formData.append("file-" + i, file);
+        if (files[0] && files[0].type === "image/gif") {
+            // we cannot crop a gif with uppy
+            formData.append("file-0", files[0]);
+        } else {
+            formData.append("file-0", upload_widget.get_edited_file());
```

Comment:
Rather than removing the previous `formData.append("file-" + i, file);` above it, why not just add a condition that verifies if `file.type` is of gif or not and performs the necessary step. You don't need to use `files[0]`.